lib/loader.js

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import fs from 'fs';
+const fsp = fs.promises;
+import https from 'https';
+import os from 'os';
+import path from 'path';
+import url from 'url';
+
+// default caching document loader for JSON-LD Context files
+function _get(url, file, resolve, reject) {
+  const options = {
+    headers: {
+      accept: 'application/ld+json'
+    }
+  };
+  https.get(url, options, res => {
+    if(res.statusCode === 301 || res.statusCode === 302) {
+      return _get(res.headers.location, file, resolve, reject);
+    } else {
+      res.pipe(file);
+      res.on('end', resolve);
+      res.on('error', reject);
+    }
+  });
+}
+
+// default caching, redirect following document loader
+export async function documentLoader(docUrl) {
+  const cacheFile = path.join(
+    os.tmpdir(), 'jsonld-context' + url.parse(docUrl)
+      .pathname.replace(/\//g, '-').replace(/\.jsonld$/, '') + '.jsonld');
+  try {
+    await fsp.access(cacheFile, fs.constants.F_OK | fs.constants.R_OK);
+  } catch(e) {
+    const file = fs.createWriteStream(cacheFile);
+    await new Promise((resolve, reject) => _get(docUrl, file, resolve, reject));
```

Comment:
I suggest moving this deferred pattern into the _get helper and then just `await _get()` here and also named parameters are preferred.

Reply:
Refactored in 4679cc14dc52b5044a885c9e01c3279f8abb0d22.